src/librustc/middle/entry.rs

```diff
@@ -163,7 +163,14 @@ fn configure_main(tcx: TyCtxt<'_>, visitor: &EntryContext<'_, '_>) -> Option<(De
             err.emit();
         } else {
```

Comment:
I think the parent `} else { // There is no main function.` should be extracted to a separate function to disturb main flow less:

```rust
fn error_there_is_no_main_function(...)
```

Reply:
Since this is the trailing part of the function I don't think that'd be all that helpful (in particular, there is no error-handling code which precedes or follows this block).

Reply:
It is helpful in the sense that I can read `error_there_is_no_main_function(..);`, quickly understand that this is the errror case and ignore / focus on that branch. Plus, planning for growth is a good idea. Less importantly, it also helpful in the sense that `fn error_there_is_no_main_function(...)` is indented one time less.

Reply:
I disagree, but will implement the requested changes to allow us to move forward.